chart/templates/_helpers.tpl

```diff
@@ -68,6 +75,22 @@ add below linux tolerations to workloads could be scheduled to those linux nodes
     - windows
 {{- end -}}
 
+
+{{- define "required-scheduling-node-selector-terms" -}}
+requiredDuringSchedulingIgnoredDuringExecution:
+           nodeSelectorTerms: {{ include "linux-node-selector-terms" . | nindent 14 }}
+{{- end -}}
+
+{{- define "node-affinity-with-default" -}}
+{{- if .Values.nodeAffinity }}
+{{- $nodeAffinity := .Values.nodeAffinity | nindent 10 -}}
+{{- $requiredSchedulingNodeSelectorTerms := include "required-scheduling-node-selector-terms" . -}}
+{{- $result := merge $nodeAffinity $requiredSchedulingNodeSelectorTerms }}
```

Comment:
Does this work at the moment? I thought you would have to do something like `(include "required-scheduling-node-selector-terms" . | fromYaml)`